server/spec/test_integration_WAPI_CANVAS.rb

```diff
@@ -0,0 +1,188 @@
+## Test WAPI module using real WSO2 API.
+
+require 'rubygems'
+require 'minitest'
+require 'minitest/autorun'
+require 'minitest/unit'
+require_relative '../WAPI'
+require_relative '../data_provider_esb'
+require 'rest-client'
+require 'logger'
+require 'yaml'
+require 'base64'
+
+require_relative 'test_helper'
+
+# To print details during the test run uncomment the TRACE=1 line
+TRACE=FalseClass
+#TRACE=1
+
+include Logging
+
+### Test WAPI use with Canvas
+
+class TestIntegrationWAPICANVAS < Minitest::Test
+
+  ## security.yml holds security configuration information for testing.
+  ## See security.yml.TEMPLATE for details.
+  ## Configurations are grouped by an arbitrary Application name and can
+  ## be loaded separately.
+
+  @@yml_file = TestHelper.findSecurityFile("security.yml")
+  @@yml = nil
+  @@config = nil
+
+  def load_yml
+    @@yml = YAML::load_file(File.open(@@yml_file))
+  end
+
+  def load_application(app_name)
+
+    logger.debug "#{__LINE__}: la: application: #{app_name}"
+
+    application = @@yml[app_name]
+    @token_server = application['token_server']
+    @api_prefix = application['api_prefix']
+    @key = application['key']
+    @secret = application['secret']
+    @token = application['token']
+    ## special uniqname is supplied for testing
+    @uniqname = application['uniqname']
+    #logger.debug "#{__LINE__}: la: token: #{@token} key: #{@key} secret: #{@secret}"
+  end
+
+  def setup
+    # by default assume that the tests will run well and don't
+    # need detailed log messages.
+
+    logger.level=TestHelper.getCommonLogLevel
+    #logger.level=Logger::ERROR
+    #logger.level=Logger::DEBUG
+
+    #@default_application_name = 'SD-QA-CANVAS'
+    @default_application_name = 'DLH-Canvas-TL'
+    #@default_application_name = 'CANVAS-ADMIN-DEV'
+    load_yml
+    load_application @default_application_name
+
+    @default_term = '2060';
+
+    a = Hash['api_prefix' => @api_prefix,
+             'key' => @key,
+             'secret' => @secret,
+             'token_server' => @token_server,
+             #'token' => '!sweet',
+             'token' => @token
+    ]
+
+    @w = WAPI.new(a)
+  end
+
+
+  ## run a request and parse result as json.  This assumes that request will work
+  ## and the result is valid json.
+  def run_and_get_json_result(url)
+    r = @w.get_request url
+    result = r.result
+    result_as_json = JSON.parse result
+    puts "url: #{url} result: "+result_as_json.inspect unless TRACE == FalseClass
+    result_as_json
+  end
+
+  ############## tests
+
+  # check that api object exists.
+  def test_canvas_api_object_exists
+    refute_nil @w
+  end
+
+
+  #### These test capabilities of tl esb poweruser API.
+
+  ## get expected data for self request
```

Comment:
looks like we can do some refactoring work for the following test cases. All have similar language statement inside the test case.